node/pkg/guardiansigner/guardiansigner_test.go

```diff
@@ -0,0 +1,97 @@
+package guardiansigner
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseSignerUri(t *testing.T) {
+	tests := []struct {
+		label        string
+		path         string
+		expectedType SignerType
+	}{
+		{label: "RandomText", path: "RandomText", expectedType: InvalidSignerType},
+		{label: "ArbitraryUriScheme", path: "arb://data", expectedType: InvalidSignerType},
+		// File
+		{label: "FileURI", path: "file://whatever", expectedType: FileSignerType},
+		{label: "FileUriNoSchemeSeparator", path: "filewhatever", expectedType: InvalidSignerType},
+		{label: "FileUriMultipleSchemeSeparators", path: "file://testing://this://", expectedType: FileSignerType},
+		{label: "FileUriTraversal", path: "file://../../../file", expectedType: FileSignerType},
+	}
+
+	for _, testcase := range tests {
+		t.Run(testcase.label, func(t *testing.T) {
+			signerType, _, err := ParseSignerUri(testcase.path)
+
+			assert.Equal(t, signerType, testcase.expectedType)
+
+			// If the signer type is Invalid, then an error should have been returned.
+			if testcase.expectedType == InvalidSignerType {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+
+		})
+	}
+}
+
+func TestFileSignerNonExistentFile(t *testing.T) {
+	nonexistentFileUri := "file://somewhere/on/disk.key"
+
+	// Attempt to generate signer using top-level generator
+	_, err := NewGuardianSignerFromUri(nonexistentFileUri, true)
+	assert.Error(t, err)
+
+	// Attempt to generate signer using NewFileSigner
+	_, keyPath, _ := ParseSignerUri(nonexistentFileUri)
+	fileSigner, err := NewFileSigner(true, keyPath)
+	assert.Nil(t, fileSigner)
+	assert.Error(t, err)
+}
+
+func TestFileSigner(t *testing.T) {
+	fileUri := "file://../query/dev.guardian.key"
+	expectedEthAddress := "0xbeFA429d57cD18b7F8A4d91A2da9AB4AF05d0FBe"
+
+	// For each file signer generation attempt, check:
+	//	That the signer returned is not nil
+	//	No error is returned
+	//	The public key returned by PublicKey(), converted to an eth address,
+	//		matches the expected address.
+
+	// Attempt to generate signer using top-level generator
+	fileSigner1, err := NewGuardianSignerFromUri(fileUri, true)
+	assert.NoError(t, err)
+	assert.NotNil(t, fileSigner1)
+	assert.Equal(t, ethcrypto.PubkeyToAddress(fileSigner1.PublicKey()).Hex(), expectedEthAddress)
```

Comment:
If these top level assertions fail, I assume the rest of the stuff below will also fail. For these setup stuff that "needs to always return correctly or the entire rest of the state is wrong", consider not using assert, but instead doing [require.NoError(t, err)](https://pkg.go.dev/github.com/stretchr/testify/require#NoError).

The [require](https://pkg.go.dev/github.com/stretchr/testify/require) testify package is for ending execution of the rest of a given test if the required assertion fails. That way you have 1 failure in the console output instead of 11 if the `NewGuardianSignerFromUri()` unexpectedly fails.